servant-client/src/Servant/Client/Internal/HttpClient.hs

```diff
@@ -68,16 +68,23 @@ import qualified Servant.Types.SourceT                    as S
 import qualified Network.HTTP.Client         as Client
 
 -- | The environment in which a request is run.
+--   The baseUrl and makeClientRequest function are used to create a `http-client` request.
+--   Cookies are then added to that request if a CookieJar is set on the environment.
+--   Finally the request is executed with the manager.
+--   The 'makeClientRequest' function can be used to modify the request to execute and set values which
+--   are not specified on a `servant` Request like the `responseTimeout` or the `redirectCount`
 data ClientEnv
   = ClientEnv
   { manager :: Client.Manager
   , baseUrl :: BaseUrl
   , cookieJar :: Maybe (TVar Client.CookieJar)
+  , makeClientRequest :: BaseUrl -> Request -> Client.Request
+  -- ^ this function can be used to customize the creation of `http-client` requests from `servant` requests. Default value: 'defaultMakeClientRequest'
```

Comment:
add also

`makeClientRequest` exists to allow overriding operational semantics e.g. `responseTimeout` *per request*.
If you need global modifications, you should use [`managerModifyRequest`](https://hackage.haskell.org/package/http-client-0.6.4/docs/Network-HTTP-Client.html#v:managerModifyRequest)

Reply:
.. and the note that possible `cookieJar` is applied after.